container-README.md

```diff
@@ -0,0 +1,87 @@
+# Kadalu Storage Management
+
+## Build Containers
+
+Builder base
+
+```
+docker build . --tag docker.io/kadalu-storage/builder -f Dockerfile.builder
+```
+
+Build node container
+
+```
+docker build . --tag docker.io/kadalu-storage/node -f Dockerfile.node
+```
+
+Build Server container
+
+```
+docker build . --tag docker.io/kadalu-storage/server -f Dockerfile.server
+```
+
+## Usage
+
+Create Kadalu network
+
+```
+docker network create kadalu
+```
+
+Run one instance of `kadalu-storage/server` container and spawn multiple instance of `kadalu-storage/node` containers as required.
+
+```
+docker run -d -v /sys/fs/cgroup/:/sys/fs/cgroup:ro --cap-add SYS_ADMIN --name kadalu-control --network kadalu kadalu-storage/server
+```
+
+Login and check if `kadalu-server` is running
+
+```
+docker exec -it kadalu-control /bin/bash
+root@e044bf61ea76:/# systemctl status kadalu-server
+● kadalu-server.service - Kadalu Storage Server
+     Loaded: loaded (/lib/systemd/system/kadalu-server.service; enabled; vendor preset: enabled)
+     Active: active (running) since Fri 2021-03-05 08:01:45 UTC; 11min ago
+   Main PID: 34 (kadalu-server)
+      Tasks: 2 (limit: 2227)
+     Memory: 1.8M
+     CGroup: /docker/e044bf61ea7657640deb13ffaf6dfcd9a2178f48c7a226f5ccd03597663213af/system.slice/kadalu-server.service
+             └─34 /usr/sbin/kadalu-server
+
+Mar 05 08:01:45 e044bf61ea76 systemd[1]: Started Kadalu Storage Server.
+Mar 05 08:01:45 e044bf61ea76 kadalu-server[34]: [development] Kemal is ready to lead at http://0.0.0.0:3000
+```
+
+```
+docker run -d -v /sys/fs/cgroup/:/sys/fs/cgroup:ro --cap-add SYS_ADMIN --name node1 --network kadalu kadalu-storage/node
+docker run -d -v /sys/fs/cgroup/:/sys/fs/cgroup:ro --cap-add SYS_ADMIN --name node2 --network kadalu kadalu-storage/node
+```
+
+Login to check if `kadalu-node` service is running
+
+```
+docker exec -it node1 /bin/bash
+root@c6c8292b9107:/# systemctl status kadalu-node
+● kadalu-node.service - Kadalu Storage node agent
+     Loaded: loaded (/lib/systemd/system/kadalu-node.service; enabled; vendor preset: enabled)
+     Active: active (running) since Fri 2021-03-05 07:49:34 UTC; 26min ago
+   Main PID: 33 (kadalu-node)
+      Tasks: 2 (limit: 2227)
+     Memory: 800.0K
+     CGroup: /docker/c6c8292b91070eec9de137ca99dc5f209d9a65dbbf3deee30d6bd8f13bd8ba6a/system.slice/kadalu-node.service
+             └─33 /usr/sbin/kadalu-node
+
+Mar 05 07:49:34 c6c8292b9107 systemd[1]: Started Kadalu Storage node agent.
+Mar 05 07:49:34 c6c8292b9107 kadalu-node[33]: [development] Kemal is ready to lead at http://0.0.0.0:3001
+```
+
+## TODO:
+
+* Add Volumes for `/var/lib/kadalu` for each Node containers
+* Add Volume for `kadalu-storage/server`(For Sqlite db and other things)
+* Set the required environment variables
+* How to communicate between containers(Self-heal daemon requires to connect to other storage nodes, kadalu-node agent needs to connect to Kadalu Server/Control plane)
```

Comment:
what happens if this is communicated using 'external' IP even though for 1 brick it can talk locally? As the config object to generate the volfile would be containing the information about the node level (read as ports etc).

Reply:
Found this method to create a network that works across the nodes. I am yet to try  https://www.youtube.com/watch?v=nGSNULpHHZc

For local containers, all are now part of same network